sys/src/9/amd64/multiboot.c

```diff
@@ -103,6 +105,18 @@ mbtypename(int type)
 	return "(unknown)";
 }
 
+static i32
+mbmoduleread(Chan *c, void *buf, i32 len, i64 off)
+{
+	if (!initrd) {
```

Comment:
I suspect this test is not really absolutely necessary. This function won't get called without the addarchfile, and that only gets added if initrd exists. But no harm done I guess.

Reply:
True. I think it was a hangover from the original multi initrd support.